book/src/external-snapshot-metadata.md

```diff
@@ -0,0 +1,89 @@
+
+
+## Alpha
+
+### Description
+This sidecar securely serves snapshot metadata to Kubernetes clients through the
+[Kubernetes SnapshotMetadata gRPC Service](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3314-csi-changed-block-tracking#the-kubernetes-snapshotmetadata-service-api).
+
+The sidecar authenticates and authorizes each Kubernetes backup application request made through the
+Kubernetes SnapshotMetadata gRPC Service API.
+It then acts as a proxy as it fetches the desired metadata from the CSI driver and
+streams it directly to the requesting application with no load on the Kubernetes API server.
+
+See ["The External Snapshot Metadata Sidecar"](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3314-csi-changed-block-tracking#the-external-snapshot-metadata-sidecar)
+section in the CSI Changed Block Tracking KEP for additional details on the sidecar.
+
+### Usage
+Backup applications, identified by their authorized ServiceAccount objects,
+directly communicate with the sidecar using the
+[Kubernetes SnapshotMetadata gRPC Service](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3314-csi-changed-block-tracking#the-kubernetes-snapshotmetadata-service-api).
+The authorization needed is described in the 
+["Risks and Mitigations"](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3314-csi-changed-block-tracking#risks-and-mitigations)
+section of the CSI Changed Block Tracking KEP.
+In particular, this requires the ability to use the Kubernetes
+[TokenRequest](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/)
+API and to access the objects required to use the service.
+
+The existence of this optional service is advertised by the presence of a
+[Snapshot Metadata Service CR](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3314-csi-changed-block-tracking#snapshot-metadata-service-custom-resource),
+named for the CSI driver that provisions the PersistentVolume and VolumeSnapshot objects involved.
+The CR contains the service endpoint and CA certificate, and an audience string for authentication.
+The backup application must use the Kubernetes
+[TokenRequest](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/)
+API with the audience string to obtain a Kubernetes authentication token for use in the
+Kubernetes SnapshotMetadata gRPC Service call.
+The backup application should establish trust for the CA certificate before making the gRPC call
+to the service endpoint.
+VolumeSnapshot metadata can be lengthy, and the Kubernetes SnapshotMetadata gRPC Service supports
+restarting an interrupted metadata request from an intermediate point in case of failure.
+
+The sidecar repository contains a
+[snapshot-metadata-lister](https://github.com/kubernetes-csi/external-snapshot-metadata/tree/master/examples/snapshot-metadata-lister)
```

Comment:
the `external-snapshot-metadata` repository does not have a `master` branch, instead it is `main` (link works, but has a redirection notification)